src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -502,9 +499,10 @@ SHAMap::onlyBelow (SHAMapTreeNode* node) const
 
     // An inner node must have at least one leaf
     // below it, unless it's the root_
-    assert (node->hasItem () || (node == root_.get ()));
+    auto leaf = static_cast<SHAMapTreeNode*>(node);
+    assert (leaf->hasItem () || (leaf == root_.get ()));
```

Comment:
If the `node` doesn't have the item, then it must be the `root_`, but `root_` is a `shared_ptr<SHAMapInnerNode>`, so if `node == root_.get()`, then `leaf` is going to be, what, a `SHAMapTreeNode*` that really points to a `SHAMapInnerNode`? And that would mean the call to `leaf->hasItem()` is going to be UB.

Right? :confused: